### PR TITLE
ticket 11 GET reviews queries done with tests

### PR DESCRIPTION
**Pull request by ElenaCKay**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -112,6 +112,91 @@ describe("GET /api/reviews", () => {
...
+                        review_img_url: expect.any(String),
+                        votes: expect.any(Number),
+                        category: "social deduction",
+                        owner: expect.any(String),
+                        comment_count: expect.any(Number),
+                    });
+                });
+            });
+    });
+    test("404: GET responds with error for an invaild sort_by", () => {
```
Comment (johnhutcheon) [inline __tests__/app.test.js]: This one should be a 400

File: __tests__/app.test.js [hunk 2]
```diff
@@ -112,6 +112,91 @@ describe("GET /api/reviews", () => {
...
+    });
+    test("404: GET responds with error for an invaild category", () => {
+        return request(app)
+            .get("/api/reviews?category=banana")
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("Invalid Category");
+            });
+    });
+    test("404: GET responds with error for an invaild order", () => {
```
Comment (johnhutcheon) [inline __tests__/app.test.js]: This one should also be a 400

File: __tests__/app.test.js [hunk 3]
```diff
@@ -112,6 +112,91 @@ describe("GET /api/reviews", () => {
...
+                    expect(review).toMatchObject({
+                        review_id: expect.any(Number),
+                        title: expect.any(String),
+                        designer: expect.any(String),
+                        review_img_url: expect.any(String),
+                        votes: expect.any(Number),
+                        category: "social deduction",
+                        owner: expect.any(String),
+                        comment_count: expect.any(Number),
+                    });
```
Comment (johnhutcheon) [inline __tests__/app.test.js:172]: Could we add another 200 test please which is a valid category query but has no reviews? This should respond with an empty array of articles. 👍

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Elena. Nice work so far. Could you just see my comments please and amend the tests? Thanks 🙂

File: __tests__/app.test.js [hunk 4]
```diff
@@ -112,6 +112,101 @@ describe("GET /api/reviews", () => {
...
+            .get("/api/reviews?sort_by=banana")
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("Invalid Sort Query");
+            });
+    });
+    test("404: GET responds with error for an invaild category", () => {
+        return request(app)
+            .get("/api/reviews?category=banana")
+            .expect(400)
```
Comment (B7Bprog) [inline __tests__/app.test.js]: Please expect a 404 here instead of a 400. This category doesn't exist, but not invalid.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Elena, please refactor that test where I commented, then please merge it.
